src/BreakpointViewer.cpp

```diff
@@ -0,0 +1,621 @@
+#include <QPainter>
+#include <QPaintEvent>
+#include <QMessageBox>
+#include <QToolTip>
+#include <QComboBox>
+#include <QDebug>
+#include <cassert>
+#include "BreakpointViewer.h"
+#include "ui_BreakpointViewer.h"
+#include "Convert.h"
+#include "CommClient.h"
+#include "OpenMSXConnection.h"
+
+enum TableColumns {
+	ENABLED = 0,
+	WP_TYPE = 1,
+	LOCATION = 2,
+	BP_ADDRESS = 2,
+	WP_REGION = 2,
+	CONDITION = 3,
+	SLOT = 4,
+	SEGMENT = 5,
+};
+
+struct ItemPos {
+	BreakpointViewer::Type type;
+	int row;
+	bool operator==(const ItemPos &ip) const
```

Comment:
(Detail)
We prefer 'ItemPos& ip' over 'ItemPos &ip'. (I'm not sure which style is used in the rest of the debugger code).